src/StackExchange.Redis/HashSlotMovedEventArgs.cs

```diff
@@ -27,7 +27,15 @@ public sealed class HashSlotMovedEventArgs : EventArgs, ICompletable
         /// </summary>
         public EndPoint NewEndPoint { get; }
 
-        internal HashSlotMovedEventArgs(EventHandler<HashSlotMovedEventArgs> handler, object sender,
+        /// <summary>
+        /// C'tor
+        /// </summary>
+        /// <param name="handler">Delegate to handle the event.</param>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="hashSlot">Hash slot.</param>
+        /// <param name="old">Old endpoint.</param>
+        /// <param name="new">New endpoint.</param>
+        public HashSlotMovedEventArgs(EventHandler<HashSlotMovedEventArgs> handler, object sender,
```

Comment:
this is a very unusual API, for specific reasons; I don't think any external caller (for unit testing etc) would need this - or indeed could use this; I wonder if the `public` constructor should just be:

``` c#
public HashSlotMovedEventArgs(int hashSlot, EndPoint old, EndPoint @new) : this (null, null, hashSlot, old, new) {}
```

thoughts?

Reply:
Still this is a public event and someone may be interested to unit-test some scenarios.

Reply:
My answer to that, then: is *show* the scenario that needs that, ideally in a test.  Adding something without a demonstrated use-case is ... not a good way to design the API usually.

Reply:
@mgravell this is an outdated comment. I've updated a pr.